modules/re2/2023-09-01/MODULE.bazel

```diff
@@ -0,0 +1,24 @@
+# Copyright 2009 The RE2 Authors.  All Rights Reserved.
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+
+# Bazel (http://bazel.build/) MODULE file for RE2.
+
+module(
+    name = "re2",
+    version = "2023-09-01",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "platforms", version = "0.0.7")
+bazel_dep(name = "rules_cc", version = "0.0.8")
+bazel_dep(name = "abseil-cpp", version = "20230802.0", repo_name = "com_google_absl")
+bazel_dep(name = "google_benchmark", version = "1.8.2", repo_name = "com_github_google_benchmark")
```

Comment:
We tried previously, but I had to revert that as per https://github.com/bazelbuild/bazel-central-registry/pull/675#issuecomment-1572561643. Has something changed in that regard since then?

Reply:
No, it's just that the list has grown and includes rulesets now. Is there a further subset of tests that you could run without e.g. Python? You could still run the full set of tests in your own CI before a release.

Reply:
`MODULE.bazel` is a bit larger now because I recently spent **_quite_** a lot of time and effort migrating to Bzlmod entirely. ;)

If this is really about keeping the presubmit lightweight, sure, I can yeet `@re2//python:all` from the build/test targets.

Reply:
This is less about CI load and more about the general cost of dependencies. Even with Bzlmod, they aren't free. For example, rules_python recently made some breaking changes to its Bzlmod API. Since re2 is, at least on the surface, mostly "just" a C++ project, users could be confused if it ends up updating their rules_python module.

If your Bzlmod migration allows you to split out these tests into a separate module in a subdirectory, you could also run them as a BCR test module.

Reply:
Bazel wouldn't fetch the Python dependencies unless the user wants the Python functionality, right? If a user depends only on `@re2//:re2`, not on `@re2//python:re2`, Bazel could (and does) prune the transitive closure of dependencies?

Reply:
Testing locally seems to corroborate what I thought: from a clean (expunged!) state, `bazel build //:re2` doesn't end up fetching pybind11 – via the `http_archive()` in pybind11_bazel – whereas `bazel build //python:re2` does.

Reply:
If we must be more ruthless about using `dev_dependency`, I can build (as opposed to build **_and_** test) `@re2//:re2` only, but that won't really do anything to change the fact that a user will (eventually) want to depend on `@re2//python:re2`. If there's some limitation of Bazel, Bzlmod or BCR that requires RE2 to have **_two_** separate modules, I can be convinced to arrange that, I suppose, but I will need a lot of guidance.

(Ping, @meteorcloudy.)

Reply:
Sounds like the python toolchain dependency is valid, but googletest and google_benchmark should still be "dev_dependency"

You can achieve this by using a [test module](https://github.com/bazelbuild/bazel-central-registry/tree/main/docs#test-module). You can even just use the re2 module as the test module, if you set `module_path: "./"`, then you can run tests as `//:all` instead of `@re2//:all` and dev dependencies will be fetched.

I'll recommend still you keep the build_and_test configure, since it actually verifies the module when it's used as a dependency.

Reply:
> Sounds like the python toolchain dependency is valid, but googletest and google_benchmark should still be "dev_dependency"

Done. Instead of tagging an RE2 release just for this, I have patched `MODULE.bazel` to set `dev_dependency = True` for `google_benchmark`, `googletest` and `abseil-py`.

> You can achieve this by using a [test module](https://github.com/bazelbuild/bazel-central-registry/tree/main/docs#test-module). You can even just use the re2 module as the test module, if you set `module_path: "./"`, then you can run tests as `//:all` instead of `@re2//:all` and dev dependencies will be fetched.

Oo! That's a very cool trick. Thanks! :D

> I'll recommend still you keep the build_and_test configure, since it actually verifies the module when it's used as a dependency.

I have split `build_and_test` such that now `build` builds the two public targets that users would depend on and then, inside the `bcr_test_module`, `test` tests everything.